Upgrade Molecule from 2.22 to 3.0.2 (#2749)
operator-framework/operator-sdk: doc/migration/version-upgrade-guide.md

                  -r "-"
```

#### Molecule Upgrade for Ansible based-operators

> The Molecule version for Ansible based-operators was upgrade from `2.22` to `3.0.2`. Following the changes required in the default scaffold files.

Reviewer: ```suggestion
The Molecule version for Ansible based-operators was upgraded from `2.22` to `3.0.2`. The following changes are required in the default scaffold files.
```

> - Remove the `scenario.name` from molecule.yaml, and ensure that any condition with in the `verify.yaml` will look for the folder name which determines the scenario name from now on

Reviewer: I'm not sure I understand this line, where is the scenario.name used within the verify.yml?

Author: Yep. I updated it .. good catch.

> - Remove the `scenario.name` from molecule.yaml, and ensure that any condition with in the `verify.yaml` will look for the folder name which determines the scenario name from now on

Reviewer: ```suggestion
- Remove the `scenario.name` from molecule.yaml, and ensure that any condition with in the `verify.yml` will look for the folder name which determines the scenario name from now on
```

- Replace the lint with newer syntax from [documentation](https://molecule.readthedocs.io/en/latest/configuration.html#lint). See:

Replace:

```yaml
lint:
  name: yamllint
  options:
    config-data:
      line-length:
        max: 120
```

With:

```yaml
lint: |
  set -e
  yamllint -d "{extends: relaxed, rules: {line-length: {max: 120}}}" .
```

```

Replace:

```yaml
lint:
  name: ansible-lint
```

With:

```yaml
lint: |
  set -e
  ansible-lint
```

> - Rename `molecule/playbook.yml` to `molecule/converge.yml` to avoid a deprecation message.

Reviewer: ```suggestion
- Rename `molecule/$SCENARIO/playbook.yml` to `molecule/$SCENARIO/converge.yml` to avoid a deprecation message.
```

> - Update the `.travis` file to install the supported lints as follows. 

Reviewer: ```suggestion
- Update the `.travis.yml` file to install the supported lints as follows. 
```
